### DISCUSSION TITLE
How to verify ownership of the site in Google Search Console?

### DISCUSSION THREAD

**Discussion by rajevv** [alshedivat/al-folio#442] [Q&A] [answered]
Hi, Thanks for this great template. I have been able to successfully set up my webpage. But I am running into problems verifying the ownership of the website in Google Search Console. My question is where should I upload the verification HTML file? I am uploading it in the root folder but it's not getting verified. Thanks.

**Comment by rohandebsarkar** [comment]
This is not currently supported directly through the `_config.yml` file (can be done with [`jekyll-seo-tag`](https://github.com/jekyll/jekyll-seo-tag)).

But you can manually verify it by either downloading the HTML file from the Google Search Console and then uploading it to the root of the repository.

OR, you can verify using the HTML tag, by copying the meta tag from the Google Search Console and pasting it to the `_includes/head.thml` file.

**Reply by mmIIrrOOss** [reply]
Where is this tag located

**Reply by rohandebsarkar** [reply]
The tag will be provided by Google Search Console while verifying the site.

**Comment by rohandebsarkar** [comment]
You can check if the uploaded file is visible by checking the file URL `username.github.io/file.html`.

I just checked your repository. It looks like the file is visible at [rajevv.github.io/google7aa70dc295e903b6.html](https://rajevv.github.io/google7aa70dc295e903b6.html). Just try after sometime, I think it will be done.

**Reply by rajevv** [reply]
Yes, it is done. Thanks for the help.

**Reply by KILLERISHERE133** [reply]
yep youre right now it works

**Comment by rohandebsarkar** [comment, marked as answer]
`al-folio` now supports the search engine verifications. You will have to just update the following lines in `_config.yml` with the respective verification IDs. 

https://github.com/alshedivat/al-folio/blob/c1428532e0735210175034d2588ce00446d3c335/_config.yml#L97-L98

**Reply by mfesgin** [reply]
> `al-folio` now supports the search engine verifications. You will have to just update the following lines in `_config.yml` with the respective verification IDs.
> 
> https://github.com/alshedivat/al-folio/blob/c1428532e0735210175034d2588ce00446d3c335/_config.yml#L97-L98

I've added the Google site verification ID (given by Google) 3 days ago, but still Google can't verify ownership. Am I supposed to provide a different URL than my main page (https://mfesgin.github.io/)?

**Reply by rohandebsarkar** [reply]
You would also have to enable `enable_google_verification`.
https://github.com/alshedivat/al-folio/blob/7172df0877a30b12feea4b6c41adfdd9da7951c0/_config.yml#L300

**Reply by mfesgin** [reply]
Thanks! I've missed that.

**Reply by giuliaguidi** [reply]
> `al-folio` now supports the search engine verifications. You will have to just update the following lines in `_config.yml` with the respective verification IDs.
> 
> https://github.com/alshedivat/al-folio/blob/c1428532e0735210175034d2588ce00446d3c335/_config.yml#L97-L98

Just IDs or also .html? Thanks!

**Reply by lucasresck** [reply]
How much time should I wait after setting up `google_site_verification` and `enable_google_verification`?

**Reply by lucasresck** [reply]
Hi, @mfesgin, you have both `google_site_verification` and `enable_google_verification` set up in your [`_config.yml`](https://github.com/mfesgin/mfesgin.github.io/blob/aff700e111376647559631e5b7cfa8a32d182bdd/_config.yml). But when I search for `mfesgin.github.io` in [Google Admin Toolbox](https://toolbox.googleapps.com/apps/dig/#TXT/) I don't find any `TXT` record. Can you confirm, please, that you were able to verify your domain property in Google Search Console using `TXT`? In the positive case, shouldn't this record appear in Google Admin Toolbox (see [domain name provider troubleshooting](https://support.google.com/webmasters/answer/9008080))?

**Reply by lucasresck** [reply]
I realized that `google_site_verification` and `enable_google_verification` adds an HTML tag to `_includes/metadata.html`: https://github.com/alshedivat/al-folio/blob/6e01a61a4ac1f39a8cef41d8bc249c54ef5187c5/_includes/metadata.html#L4

Therefore, Google Search Console property type is **URL prefix** and **not domain**.

Domain verification is the first (and more desirable) option in Google Search Console, what explains my confusion. It shouldn't be the case for my homepage to appear in [Google Admin Toolbox](https://toolbox.googleapps.com/apps/dig/#TXT/).

**Reply by LiuYing-1** [reply]
Hi, Lucas! Do I need to upload the downloaded file from URL prefix option online? Where the file should be in the project?

**Reply by lucasresck** [reply]
Hi, @LiuYing-1, you just need to place the downloaded HTML file to the repository's root. Take a look at [how I did](https://github.com/lucasresck/lucasresck.github.io/blob/master/googledddcdc5eee18da83.html). I can confirm it was correctly identified by Google Search Console.

Important to notice that this approach is different than the URL prefix HTML tag I pointed in my last comment. In any case, it's good to have more than one verification method active.

**Reply by Abaadireeammannuur** [reply]
Faayila isinii erguu
